src/client/Microsoft.Identity.Client/Internal/Requests/Silent/CacheSilentStrategy.cs

```diff
@@ -153,6 +154,8 @@ private async Task<AuthenticationResult> CreateAuthenticationResultAsync(MsalAcc
         {
             var msalIdTokenItem = await CacheManager.GetIdTokenCacheItemAsync(cachedAccessTokenItem).ConfigureAwait(false);
             var tenantProfiles = await CacheManager.GetTenantProfilesAsync(cachedAccessTokenItem.HomeAccountId).ConfigureAwait(false);
+            var accounts = await CacheManager.GetAccountsAsync().ConfigureAwait(false);
+            var account = accounts.FirstOrDefault() as Account;
```

Comment:
Have a look at `CacheManager.GetIdTokenCacheItemAsync(accessTokenItem)` - this gets an IdToken from the cache, using the data of the AT (env, home account id, tenant id etc.). We should have the same logic for account. 

You can go a step further and combine `GetTenantProfilesAsync` and this new `GetAccountAssociatedWithAccessToken` as well, because an Account object can be materialized from AccountCacheItem + tenant profiles.

Reply:
Changed